skimage/restoration/_inpaint.pyx

```diff
@@ -0,0 +1,72 @@
+import numpy as np
+import cython
+
+cimport numpy as cnp
+from .._shared.fused_numerics cimport np_floats
+
+
+ # Can't use "const np_floats[::1]", but instead must declare fused const types
+ # See: https://github.com/cython/cython/issues/1772
+
+cdef fused const_float_1d_c:
+    const float[::1]
+    const double[::1]
+
+cdef fused const_floating_2d_c:
+    const float[:, ::1]
+    const double[:, ::1]
```

Comment:
That would work, but I now realize that we may want to just not declare them const to avoid additional unneeded variants of the function from being generated. We really only need two variants: one for float32 and one for float64. However since `np_floats` and `const_floats` each fuse two types, we get C functions generated for all 4 possible combinations. Example:

```C
static Py_ssize_t __pyx_fuse_0_0__pyx_f_7skimage_11restoration_8_inpaint__build_matrix_inner(Py_ssize_t, Py_ssize_t, Py_ssize_t, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, int __pyx_skip_dispatch); /*proto*/
static Py_ssize_t __pyx_fuse_0_1__pyx_f_7skimage_11restoration_8_inpaint__build_matrix_inner(Py_ssize_t, Py_ssize_t, Py_ssize_t, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, int __pyx_skip_dispatch); /*proto*/
static Py_ssize_t __pyx_fuse_1_0__pyx_f_7skimage_11restoration_8_inpaint__build_matrix_inner(Py_ssize_t, Py_ssize_t, Py_ssize_t, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, int __pyx_skip_dispatch); /*proto*/
static Py_ssize_t __pyx_fuse_1_1__pyx_f_7skimage_11restoration_8_inpaint__build_matrix_inner(Py_ssize_t, Py_ssize_t, Py_ssize_t, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, __Pyx_memviewslice, int __pyx_skip_dispatch); /*proto*/
```

If I just used `np_floats` everywhere this would not happen.